src/main/java/ch/njol/skript/effects/EffOperations.java

```diff
@@ -0,0 +1,235 @@
+	+ "Literals cannot be used on the left-hand side.")
+@Example("""
+	set {_num} to 1
+	multiply {_num} by 10
+	divide {_num} by 5
+	raise {_num} to the power of 2
+	""")
+@Example("""
+	set {_nums::*} to 15, 21 and 30
+	divide {_nums::*} by 3
+	multiply {_nums::*} by 5
+	raise {_nums::*} to the power of 3
+	""")
+@Example("""
+	set {_vector} to vector(1,1,1)
+	multiply {_vector} by vector(4,8,16)
+	divide {_vector} by 2
+	""")
+@Example("""
+	set {_timespan} to 1 hour
+	multiply {_timespan} by 3
+	""")
+@Example("""
+	# Will error due to literal
+	multiply 1 by 2
+	divide 10 by {_num}
+	""")
+@Since("INSERT VERSION")
+public class EffOperations extends Effect implements SyntaxRuntimeErrorProducer {
+
+	private static final Patterns<Operator> PATTERNS = new Patterns<>(new Object[][]{
+		{"multiply %~objects% by %object%", Operator.MULTIPLICATION},
+		{"divide %~objects% by %object%", Operator.DIVISION},
+		{"raise %~objects% to [the] (power|exponent) [of] %object%", Operator.EXPONENTIATION}
+	});
+
+	static {
+		Skript.registerEffect(EffOperations.class, PATTERNS.getPatterns());
+	}
+
+	private Operator operator;
+	private Expression<?> left;
+	private Class<?>[] leftAccepts;
+	private Expression<?> right;
+	private Node node;
+	private Operation<Object, Object, Object> operation = null;
+	private OperationInfo<?, ?, ?> operationInfo;
+
+	@Override
+	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		operator = PATTERNS.getInfo(matchedPattern);
+		node = getParser().getNode();
+		left = exprs[0];
+		right = LiteralUtils.defendExpression(exprs[1]);
+
+		leftAccepts = left.acceptChange(ChangeMode.SET);
+		// Ensure 'left' is changeable
+		if (leftAccepts == null) {
+			Skript.error("'" + left + "' cannot be set to anything and therefore cannot be " + getOperatorName() + ".");
+			return false;
+		} else if (leftAccepts.length == 0) {
+			throw new IllegalStateException("An expression should never return an empty array for a ChangeMode of 'SET'");
+		}
+		// Ensure the accepted classes of 'left' are non-array classes
+		for (int i = 0; i < leftAccepts.length; i++) {
+			if (leftAccepts[i].isArray()) {
+				leftAccepts[i] = leftAccepts[i].getComponentType();
+			}
+		}
+
+		Class<?> leftType = left.getReturnType();
+		Class<?> rightType = right.getReturnType();
+
+		if (leftType.isArray())
+			leftType = leftType.getComponentType();
+
+		if (leftType.equals(Object.class) && rightType.equals(Object.class)) {
+			// 'left' and 'right' return 'Object.class' thus making operation checks non-applicable
+			// However, we can check to make sure any of the registered operations return types are applicable
+			// 		for 'left's acceptedClasses
+			Class<?>[] allReturnTypes = Arithmetics.getAllReturnTypes(operator).toArray(Class[]::new);
+			if (!ChangerUtils.acceptsChangeTypes(leftAccepts, allReturnTypes)) {
+				Skript.error(left + " cannot be " + getOperatorName() + ".");
```

Comment:
```suggestion
				Skript.error(left.toString(null, Skript.debug()) + " cannot be " + getOperatorName() + ".");
```
It is not guaranteed that regular `toString` will return a readable result for an expression.